src/generate.py

```diff
@@ -296,6 +296,7 @@ def generate_C_free(obj, c_file):
                 c_file.write("    if (ptr->%s)\n" % (i.origname))
                 c_file.write("        free_%s (ptr->%s);\n" % (typename, i.origname))
                 c_file.write("    ptr->%s = NULL;\n" % (i.origname))
+    c_file.write("    free(ptr);\n")
```

Comment:
please leave an empty space between 'free' and '(ptr)' so we generate similar code